### PR TITLE
Fix for BC breaking changes

### PR DESCRIPTION
**Pull request by lindelius**
Fix for BC breaking changes to `AutoloadGenerator` introduced in https://github.com/composer/composer/pull/11663

### CODE REVIEW

**Comment by drupol** [comment]
Sorry about that colleagues!

**Comment by lindelius** [comment]
@beberlei would be great to get this merged, as the plugin is no longer usable after upgrading Composer to version 2.6.4

**Comment by fredden** [comment]
You should probably also increase the minimum supported version of `composer-plugin-api`, as this change won't work on older versions of Composer.
https://github.com/stockfiller/composer-monorepo-plugin/blob/e2b73cbee0cea00900e716ff30bcea203d51a1ab/composer.json#L6

**Comment by beberlei** [comment]
For some reason the checks also did not run, do the tests pass?

And yes minimum versions need to be adjusted for plugin api and composer itself if there is a dev constraint

**Comment by Seldaek** [comment]
Sorry about this, no need to change anything I'll fix this in 2.6.5, I thought of moving this to the end for safety but then it got lost at some point, and now here we are 😅

File: src/main/Monorepo/Build.php [hunk 1]
```diff
@@ -90,6 +90,7 @@ public function build($rootDirectory, $optimize = false, $noDevMode = false, $cl
                 $localRepo,
                 $mainPackage,
                 $installationManager,
+                $composer->getLocker(),
                 'composer',
                 $optimize
```
Comment (Seldaek) [inline src/main/Monorepo/Build.php]: See https://github.com/composer/composer/commit/cb363b0e84459b8d280d0ff6b5a7d4ccde93420d

You can do this to be compatible with all Composer versions except 2.6.4 which will break but I think that's ok as it was breaking anyway.

```suggestion
                'composer',
                $optimize,
                null,
                $composer->getLocker()
```

**Comment by drupol** [comment]
Ah good idea. I can provide the fix if you want... since I introduced the issue myself, I actually don't mind doing it.

**Comment by Seldaek** [comment]
It's done https://github.com/composer/composer/commit/cb363b0e84459b8d280d0ff6b5a7d4ccde93420d

**Comment by drupol** [comment]
Excellent, looking forward for 2.6.5 ! :)

File: composer.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     },
     "require-dev": {
         "php": "^5.5|^7.0|^8.0",
-        "composer/composer": "^2.0.7",
+        "composer/composer": "^2.6.4",
```
Comment (drupol) [inline composer.json:10]: How about using `^2.6.5` and remove the `if` condition down here?

Comment (lindelius) [reply composer.json:10]: Sure, if that's preferred, then we can do that as soon as we've verified that all of our environments and containers have been updated

**Pull request closed without merging**